pentagon/component/__init__.py

```diff
@@ -0,0 +1,53 @@
+import os
+import jinja2
+import glob
+import shutil
+import logging
+import traceback
+import sys
+from pentagon.helpers import render_template
+
+
+class ComponentBase():
+    """ Base class for Pentagon Components. """
+    _required_parameters = []
+
+    def __init__(self, data, additional_args=None, **kwargs):
+
+        self._data = data
+        self._additional_args = additional_args
+
+        missing_parameters = []
+        for item in self._required_parameters:
+            if item not in self._data.keys():
+                missing_parameters.append(item)
+
+        if missing_parameters:
+            logging.error("Missing required data parameters: {}".format(",".join(missing_parameters)))
+            sys.exit(1)
+
+    @property
+    def _destination_directory_name(self):
+        if self._destination != './':
+            return self._destination
+        return self._data.get('name', self.__class__.__name__.replace('.', '_').lower())
+
+    @property
+    def _files_directory(self):
+        return sys.modules[self.__module__].__path__[0] + "/files"
+
+    def add(self, destination):
+        """ Copies files and templates from <component>/files and templates the *.jinja files """
+
+        self._destination = destination
+        try:
+            shutil.copytree(self._files_directory, self._destination_directory_name)
+            for template in glob.glob(self._destination_directory_name + "/*.jinja"):
+                name = template.split('/')[-1]
+                path = '/'.join(template.split('/')[0:-1])
+                target = self._destination_directory_name + "/" + name.replace(".jinja", "")
```

Comment:
This is likely an edge-case that we won't run into often, this looks like it will replace `jinja` anywhere in the string so something like `jinja-template.jinja` would turn into `-template`. I don't think it needs to be fixed for our use case.

Reply:
Thats a good point. I'd like to sort it out now before we forget about it and are very confused.